kano_updater/status.py

```diff
@@ -125,12 +127,20 @@ def save(self):
         with open(self._status_file, 'w') as status_file:
             json.dump(data, status_file, indent=4)
 
-        # ensure splash is set whenever we are installing and not otherwise
-        if self._state in [self.INSTALLING_UPDATES,
-                           self.INSTALLING_INDEPENDENT]:
+        # When installing updates, configure the recovery stategy for the next
+        # boot in case of power failure. This sets a different splash screen
+        # during bootup and configures the system to autologin as the user in
+        # order to start the Updater immediately. See kano-ui-autostart.
+        if self._state in [self.INSTALLING_UPDATES, self.INSTALLING_INDEPENDENT]:
+            logger.debug('Configuring recovery stategy for next boot')
             set_splash_interrupted()
+            enable_console_autologin(get_user_unsudoed())
+            set_ldm_autologin(get_user_unsudoed())
+            enable_ldm_autostart()
```

Comment:
Can we add a new function to `kano_init` for `set_user_autologin()` (or something like that) which does these three functions in order; the updater knows too much about the internals of the login process here: these shouldn't be part of the public API.